tdct/deconvolution.py

```diff
@@ -0,0 +1,864 @@
+
+    normaliseinputs=False
+
+    #Read data
+    if os.path.isfile(datapath) is True and os.path.isfile(psfdatapath) is True and niter>0:
+        if debug is True: print(clrmsg.DEBUG, "Loading images: ", datapath," , ",psfdatapath)
+
+        data_np = tf.imread(datapath)
+        psf_np = tf.imread(psfdatapath)
+
+        progr0.increment() #1
+
+        #Convert and normalise
+        data_np_norm = _convertAndNormalise(data_np,normaliseinputs)
+        psf_np_norm = _convertAndNormalise(psf_np, normaliseinputs)
+
+        s1 = data_np_norm.shape
+        s2 = psf_np_norm.shape
+
+        shape = [(s1[i] + s2[i] - 1) for i in range(data_np_norm.ndim)]
+        fshape = [scipy.fft.next_fast_len(shape[a], True) for a in range(len(shape))]
+
+        #scipy.fft routines need to axes to be provided otherwise they only do FFT in last axes
+        #https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.rfftn.html
+        axes = [i for i in range(data_np_norm.ndim)]
+
+        progr0.increment() #2
+
+        #Precalculated psf_fft
+        psf_fft= scipy.fft.rfftn(psf_np_norm,fshape, axes)
+
+        #and precalculate psf_flip_fft
+        psf_flip = np.flip(psf_np_norm)
+        psf_flip_fft = scipy.fft.rfftn(psf_flip,fshape, axes)
+
+        #Precalculate fslice (used to fix image sizes after using fast_len method)
+        fslice = tuple([slice(sz) for sz in shape])
+
+        progr0.increment() #3
+
+        xn1 = np.array(data_np_norm) #initialize copy
+        for i in range(niter):
+            xn=xn1
+
+            #denominator convolution
+            xn_fft = scipy.fft.rfftn(xn,fshape, axes)
+            ret = scipy.fft.irfftn(xn_fft * psf_fft, fshape, axes)
+            #using fast_len, so fix sizes
+            Hx0 = ret[fslice]
+            Hx = _centered(Hx0, s1).copy() #Fix shape as in mode='same'
+
+            yhx = np.divide(data_np_norm,Hx)
+
+            progr0.increment()
+
+            #2nd convolution
+            yhx_fft = scipy.fft.rfftn(yhx,fshape, axes)
+            ret = scipy.fft.irfftn(yhx_fft * psf_flip_fft, fshape, axes)
+            #using fast_len, so fix sizes
+            htyhx0 = ret[fslice]
+            htyhx = _centered(htyhx0, s1).copy() #Fix shape as in mode='same'
+
+            xn1 = np.multiply(xn, htyhx)
+
+            progr0.increment()
+
+        data_deconv_norm_256 = _convertAndNormalise(xn1)*256
+        data_deconv_uint8 = data_deconv_norm_256.astype('uint8')
+
+        progr0.increment() #4
+        
+        #Save data
+        fpath,fname = os.path.split(datapath)
+        fname0 = os.path.join(fpath, "DeconvRL"+str(niter)+"it_"+fname)
+        tf.imsave(fname0 ,data_deconv_uint8)
+
+        progr0.setmax()
+ 
+        if debug is True: print(clrmsg.DEBUG, "Completed deconvolution, file saved: ", fname0)
+
+
+#An implementation of the circular() routine from DeconvolutionLab2 that shifts the data centre of the psf
+def _circulify3D(data3D):
```

Comment:
Again, I don't think it's necessary to have functions with starting "_". Also put the comment in a doc string instread. This is helpful because you can call the python function >>> help(_circulify3D) and it will print the docstring!